<?xml version="1.0"?>
<robot xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor"
       xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller"
       xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface">

  <property name="M_PI" value="3.1415926535897931" />
 
  <macro name="m3_l_chassis" params="side parent reflect *origin">

  <joint name="l_chassis_joint" type="fixed">
    <origin xyz="0 0 -0.04414" rpy="0 0 0" />
    <parent link="${parent}"/>
    <child link="l_chassis" />
  </joint>

     <link name="l_chassis">
      <inertial>
        <mass value="0.663" />
	<origin xyz="0.001459 0.001619 -0.051312"/>
	<inertia ixx="0.000873"
	iyy="0.000597"
	izz="0.000565"
	ixy="0.000031"
	ixz="-0.000008"
	iyz="0.000016" />
      </inertial>

      <visual>
	      <material name="GreyA" />
	      <origin xyz="0 0 0" rpy="0 0 0 " />
	      <geometry name="l_chassis_geom">
		      <mesh filename="package://${MESH}/G2R1/chassis.dae" />
	      </geometry>  
      </visual>

      <collision>
	      <origin xyz="0 0 0" rpy="0.0 0.0 0.0 " />
	      <geometry name="l_chassis_collision">
		      <box size="0.01 0.01 0.01" />
	      </geometry>
      </collision>
    </link>

    <joint name="left_gripper_j0" type="revolute">
      <axis xyz="1 0 0" />
      <origin xyz="0 0.021 -0.09013" rpy="0 0 0" />
      <parent link="l_chassis"/>
      <child link="l_base_0"/>
      <limit effort="100" velocity="${20.0*M_PI/180}" lower="${-40*M_PI/180}" upper="${15*M_PI/180}" />
    </joint>

    <link name="l_base_0">
      <inertial>
        <mass value="0.0117" />
	<origin xyz="0.0 0.0 0.0"/>
	<inertia ixx="0.0"
	iyy="0.0"
	izz="0.0"
	ixy="0.0"
	ixz="0.0"
	iyz="0.0" />
      </inertial>
      <visual>
	      <material name="GreyA" />
	      <origin xyz="0 0 0" rpy="0 0 0 " />
	      <geometry name="l_base0_geom">
		      <mesh filename="package://${MESH}/G2R1/base.dae" />
	      </geometry>  
      </visual>
      <collision>
	      <origin xyz="0 0 0" rpy="0.0 0.0 0.0 " />
	      <geometry name="l_base0_collision">
		      <box size="0.01 0.01 0.01" />
	      </geometry>
      </collision>
    </link>

    <joint name="left_gripper_j1" type="revolute">
      <axis xyz="1 0 0" />
      <origin xyz="0 -0.021 -0.09013" rpy="0 0 0" />
      <parent link="l_chassis"/>
      <child link="l_base_1"/>
      <limit effort="100" velocity="${20.0*M_PI/180}" lower="${-40*M_PI/180}" upper="${15*M_PI/180}" />
    </joint>

    <link name="l_base_1">
      <inertial>
        <mass value="0.0117" />
	<origin xyz="0.0 0.0 0.0"/>
	<inertia ixx="0.0"
	iyy="0.0"
	izz="0.0"
	ixy="0.0"
	ixz="0.0"
	iyz="0.0" />
      </inertial>
      <visual>
	      <material name="GreyA" />
	      <origin xyz="0 0 0" rpy="0 0 3.14 " />
	      <geometry name="l_base1_geom">
		      <mesh filename="package://${MESH}/G2R1/base.dae" />
	      </geometry>  
      </visual>
      <collision>
	      <origin xyz="0 0 0" rpy="0.0 0.0 0.0 " />
	      <geometry name="l_base1_collision">
		      <box size="0.01 0.01 0.01" />
	      </geometry>
      </collision>
    </link>
    
     <joint name="left_gripper_j2" type="revolute">
      <axis xyz="1 0 0" />
      <origin xyz="0 0.0 -0.055" rpy="0 0 0" />
      <parent link="l_base_0"/>
      <child link="l_tip_0"/>
      <limit effort="100" velocity="${20.0*M_PI/180}" lower="${-40*M_PI/180}" upper="${15*M_PI/180}" />
    </joint>

    <link name="l_tip_0">
      <inertial>
        <mass value="0.0117" />
	<origin xyz="0.0 0.0 0.0"/>
	<inertia ixx="0.0"
	iyy="0.0"
	izz="0.0"
	ixy="0.0"
	ixz="0.0"
	iyz="0.0" />
      </inertial>
      <visual>
	      <material name="GreyA" />
	      <origin xyz="0 0 0" rpy="0 0 3.14 " />
	      <geometry name="l_tip0_geom">
		      <mesh filename="package://${MESH}/G2R1/tip.dae" />
	      </geometry>  
      </visual>
      <collision>
	      <origin xyz="0 0 0" rpy="0.0 0.0 0.0 " />
	      <geometry name="l_tip0_collision">
		      <box size="0.01 0.01 0.01" />
	      </geometry>
      </collision>
    </link>
    
    <joint name="left_gripper_j3" type="revolute">
      <axis xyz="1 0 0" />
      <origin xyz="0 0.0 -0.055" rpy="0 0 0" />
      <parent link="l_base_1"/>
      <child link="l_tip_1"/>
      <limit effort="100" velocity="${20.0*M_PI/180}" lower="${-40*M_PI/180}" upper="${15*M_PI/180}" />
    </joint>

    <link name="l_tip_1">
      <inertial>
        <mass value="0.0117" />
	<origin xyz="0.0 0.0 0.0"/>
	<inertia ixx="0.0"
	iyy="0.0"
	izz="0.0"
	ixy="0.0"
	ixz="0.0"
	iyz="0.0" />
      </inertial>
      <visual>
	      <material name="GreyA" />
	      <origin xyz="0 0 0" rpy="0 0 0 " />
	      <geometry name="l_tip1_geom">
		      <mesh filename="package://${MESH}/G2R1/tip.dae" />
	      </geometry>  
      </visual>
      <collision>
	      <origin xyz="0 0 0" rpy="0.0 0.0 0.0 " />
	      <geometry name="l_tip1_collision">
		      <box size="0.01 0.01 0.01" />
	      </geometry>
      </collision>
    </link>

    
</macro>

  <macro name="m3_l_gripper" params="side parent reflect *origin">
    <m3_l_chassis side="${side}" reflect="${reflect}" parent="${parent}">
      <insert_block name="origin" />
    </m3_l_chassis>
  </macro>


</robot>
